app/scripts/contentscript.js

```diff
@@ -82,6 +80,52 @@ function injectScript(content) {
   }
 }
 
+/**
+ * SERVICE WORKER LOGIC
+ */
+
+const WORKER_KEEP_ALIVE_INTERVAL = 1000;
+const WORKER_KEEP_ALIVE_MESSAGE = 'WORKER_KEEP_ALIVE_MESSAGE';
+const TIME_45_MIN_IN_MS = 45 * 60 * 1000;
+
+/**
+ * Don't run the keep worker alive logic for initial JSON RPC methods.
+ * We should not run keep-alive logic for non-dapp pages.
+ */
+const IGNORE_INIT_METHODS_FOR_KEEP_ALIVE = [
+  MESSAGE_TYPE.GET_PROVIDER_STATE,
+  MESSAGE_TYPE.SEND_METADATA,
+];
+
+let keepAliveInterval;
+let keepAliveTimer;
+
+/**
+ * Running this method will ensure the service worker is kept alive for 45 minutes
+ * First message is sent immediately, subsequent messages are sent at interval of WORKER_KEEP_ALIVE_INTERVAL
+ */
+const runWorkerKeepAliveInterval = () => {
+  if (keepAliveTimer) {
+    clearTimeout(keepAliveTimer);
+  }
+
+  keepAliveTimer = setTimeout(() => {
+    clearInterval(keepAliveInterval);
+  }, TIME_45_MIN_IN_MS);
+
+  if (keepAliveInterval) {
+    clearInterval(keepAliveInterval);
+  }
```

Comment:
`if` above and line 108 looks redundant.

Reply:
one is for the interval, and the other is for the timer

Reply:
I think 
```
if (keepAliveInterval) {
    clearInterval(keepAliveInterval);
  }
```

Or just this `clearInterval(keepAliveInterval);`

will be same.

Reply:
updated [ae3c2df](https://github.com/MetaMask/metamask-extension/pull/16075/commits/ae3c2dff4d60efa06bcba80799f8133f3d90493f)